homeassistant/components/blue_current/button.py

```diff
@@ -0,0 +1,91 @@
+"""Support for Blue Current buttons."""
+
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from typing import Any
+
+from bluecurrent_api.client import Client
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import Connector
+from .const import DOMAIN
+from .entity import ChargepointEntity
+
+
+@dataclass(kw_only=True, frozen=True)
+class ChargePointButtonEntityDescription(ButtonEntityDescription):
+    """Describes a Blue Current button entity."""
+
+    function: Callable[[Client, str], Coroutine[Any, Any, None]]
+
+
+CHARGE_POINT_BUTTONS = (
+    ChargePointButtonEntityDescription(
+        key="reset",
+        translation_key="reset",
+        function=lambda client, evse_id: client.reset(evse_id),
+        device_class=ButtonDeviceClass.RESTART,
+    ),
+    ChargePointButtonEntityDescription(
+        key="reboot",
+        translation_key="reboot",
+        function=lambda client, evse_id: client.reboot(evse_id),
+        device_class=ButtonDeviceClass.RESTART,
```

Comment:
This sensor can omit its translation key as the device class will take case of the icon and the name

Reply:
I would like to use the restart-alert icon for the reboot button and keep the names (both reset and reboot) so that they are named the same as on my.bluecurrent.nl.

Reply:
Should I remove the device class?